### PR TITLE
Implement AddTrajectoryHandlerTest.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer_grpc/framework/testing/rpc_handler_test_server.h [hunk 1]
```diff
@@ -87,6 +87,10 @@ class RpcHandlerTestServer : public Server {
     WaitForHandlerCompletion(RpcHandlerWrapper<RpcHandlerType>::ON_FINISH);
   }
 
+  const typename RpcHandlerType::ResponseType &response() {
```
Comment (pifon2a) [inline cartographer_grpc/framework/testing/rpc_handler_test_server.h]: nit: RpcHandlerType::ResponseType& response()

Comment (pifon2a) [inline cartographer_grpc/framework/testing/rpc_handler_test_server.h]: Out of curiosity: It should compile without "typename" everywhere, shouldn't it?

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**